### PR TITLE
Final merge of touchpad emulator code

### PR DESCRIPTION
**Pull request by dylwhich**
### Description

A couple commits I hadn't pushed into the `emu-ext-touch` branch before which clean up the window layout slightly, fix an X memory leak in rawdraw, and initialize the extensions more cleanly.

Mainly though, it finishes up the merge and makes all the emulator touch code match the new nomenclature.

### Readiness Checklist
- [x] I have run `make format` to format the changes
- [ ] I have compiled the firmware and the changes have no warnings
- [x] I have compiled the emulator and the changes have no warnings
- [ ] I have run `make cppcheck` and checked that `cppcheck_result.txt` has no warnings for the changes
- [x] I have added doxygen comments to any code used by more than one Swadge mode. This includes `/*! \file` comments with Design Philosophy, Usage, and Example sections for new headers.
- [ ] I have run `make docs` and checked that `doxy_warnings.txt` has no warnings for the new code

### CODE REVIEW

File: emulator/src/rawdraw_sf.h [hunk 1]
```diff
@@ -5020,11 +5020,13 @@ void CNFGGLXSetup( )
 		exit( -1 );
 	}
 	CNFGGLXFBConfig = cfgs[0];
+	XFree(cfgs);
 	XVisualInfo * vis = glXGetVisualFromFBConfig( CNFGDisplay, CNFGGLXFBConfig );
 	CNFGVisual = vis->visual;
 	CNFGVisualID = vis->visualid;
 	CNFGDepth = vis->depth;
 	CNFGCtx = glXCreateContext( CNFGDisplay, vis, NULL, True );
+	XFree(vis);
```
Comment (AEFeinstein) [inline emulator/src/rawdraw_sf.h:5029]: This isn't the right way to make rawdraw changes. You should make them upstream at https://github.com/cntools/rawdraw/, including regenerating the single file header, have @cnlohr merge it, and copy the whole header back here. It's a hassle, but that's open source 🙂

**Review by AEFeinstein — CHANGES_REQUESTED** [review]

File: emulator/src/rawdraw_sf.h (hunk 1 above)
Comment (dylwhich) [reply emulator/src/rawdraw_sf.h:5029]: Boo, I guess I'll do it the right way... Looks like one of these was already in there already actually.

Comment (dylwhich) [reply emulator/src/rawdraw_sf.h:5029]: In progress: https://github.com/cntools/rawdraw/pull/102

Comment (AEFeinstein) [reply emulator/src/rawdraw_sf.h:5029]: Ah, I see one was already done. Neat.

Comment (AEFeinstein) [reply emulator/src/rawdraw_sf.h:5029]: Yeah, when that's through update rawdraw here with the hash and all and we should be good to go.

**Comment by AEFeinstein** [comment]
This was a bear to fix CI for but I got there 😟

**Pull request merged**